### ISSUE TITLE
官方dlrm的训练脚本为什么会出现 loss NAN 的问题

### ISSUE THREAD

**Issue by kanyyy** [Oneflow-Inc/oneflow#10321]
## Summary

使用dlrm官方脚本。
以下是我的执行命令：

```
python3 -m oneflow.distributed.launch \
    --nproc_per_node 2 \
    --nnodes 1 \
    --node_rank 0 \
    --master_addr 127.0.0.1 \
    dlrm_train_eval.py \
      --data_dir /path/to/criteo_spark/parquet \
      --persistent_path /path/to/persistent
```
输入的训练集是由tools下的官方转换脚本处理的。训练流程能跑完，但是出现loss Nan

## Code to reproduce bug

以下是部分输出信息：
```
------------------------ arguments ------------------------
  amp ............................................. False
  bottom_mlp ...................................... [512, 256, 128]
  cache_memory_budget_mb .......................... 8192
  data_dir ........................................ /mae_pat/data/criteo_spark/parquet
  decay_batches ................................... 27772
  decay_start ..................................... 49315
  disable_dense_input_padding ..................... False
  disable_fusedmlp ................................ False
  disable_interaction_padding ..................... False
  embedding_vec_size .............................. 128
  eval_batch_size ................................. 55296
  eval_batches .................................... 1612
  eval_interval ................................... 10000
  interaction_itself .............................. False
  learning_rate ................................... 24
  loss_print_interval ............................. 1000
  loss_scale_policy ............................... static
  model_load_dir .................................. None
  model_save_dir .................................. None
  one_embedding_key_type .......................... int64
  persistent_path ................................. /mae_pat/int.jiatao/oneflow_demo_draft
  save_initial_model .............................. False
  save_model_after_each_eval ...................... False
  store_type ...................................... cached_host_mem
  table_size_array ................................ [39884406, 39043, 17289, 7420, 20263, 3, 7120, 1543, 63, 38532951, 2953546, 403346, 10, 2208, 11938, 155, 4, 976, 14, 39979771, 25641295, 39664984, 585935, 12972, 108, 36]
  top_mlp ......................................... [1024, 1024, 512, 256]
  train_batch_size ................................ 55296
  train_batches ................................... 75000
  warmup_batches .................................. 2750
-------------------- end of arguments ---------------------
/App/conda/envs/conda_gpu/lib/python3.9/site-packages/petastorm/reader.py:465: UserWarning: shard_seed was deprecated and will be removed in future versions. Use seed to apply randomization effects on sharding row groups.
  warnings.warn("shard_seed was deprecated and will be removed in future versions. "
/App/conda/envs/conda_gpu/lib/python3.9/site-packages/petastorm/reader.py:465: UserWarning: shard_seed was deprecated and will be removed in future versions. Use seed to apply randomization effects on sharding row groups.
  warnings.warn("shard_seed was deprecated and will be removed in future versions. "
/App/conda/envs/conda_gpu/lib/python3.9/site-packages/petastorm/reader.py:465: UserWarning: shard_seed was deprecated and will be removed in future versions. Use seed to apply randomization effects on sharding row groups.
  warnings.warn("shard_seed was deprecated and will be removed in future versions. "
/App/conda/envs/conda_gpu/lib/python3.9/site-packages/petastorm/reader.py:465: UserWarning: shard_seed was deprecated and will be removed in future versions. Use seed to apply randomization effects on sharding row groups.
  warnings.warn("shard_seed was deprecated and will be removed in future versions. "
/App/conda/envs/conda_gpu/lib/python3.9/site-packages/petastorm/reader.py:465: UserWarning: shard_seed was deprecated and will be removed in future versions. Use seed to apply randomization effects on sharding row groups.
  warnings.warn("shard_seed was deprecated and will be removed in future versions. "
/App/conda/envs/conda_gpu/lib/python3.9/site-packages/petastorm/reader.py:465: UserWarning: shard_seed was deprecated and will be removed in future versions. Use seed to apply randomization effects on sharding row groups.
  warnings.warn("shard_seed was deprecated and will be removed in future versions. "
/App/conda/envs/conda_gpu/lib/python3.9/site-packages/petastorm/reader.py:465: UserWarning: shard_seed was deprecated and will be removed in future versions. Use seed to apply randomization effects on sharding row groups.
  warnings.warn("shard_seed was deprecated and will be removed in future versions. "
/App/conda/envs/conda_gpu/lib/python3.9/site-packages/petastorm/reader.py:465: UserWarning: shard_seed was deprecated and will be removed in future versions. Use seed to apply randomization effects on sharding row groups.
  warnings.warn("shard_seed was deprecated and will be removed in future versions. "
W20230820 16:22:50.974692 798419 cuda_stream.cpp:48] Runtime version 11.10 of cuBLAS incompatible with compiletime version 11.11.
W20230820 16:22:51.059744 798418 cuda_stream.cpp:48] Runtime version 11.10 of cuBLAS incompatible with compiletime version 11.11.
W20230820 16:22:51.101035 798396 cuda_stream.cpp:48] Runtime version 11.10 of cuBLAS incompatible with compiletime version 11.11.
W20230820 16:22:51.221477 798419 cuda_stream.cpp:48] Runtime version 8.4 of cuDNN incompatible with compiletime version 8.6.
W20230820 16:22:51.235112 798395 cuda_stream.cpp:48] Runtime version 11.10 of cuBLAS incompatible with compiletime version 11.11.
W20230820 16:22:51.284055 798435 cuda_stream.cpp:48] Runtime version 11.10 of cuBLAS incompatible with compiletime version 11.11.
W20230820 16:22:51.285319 798393 cuda_stream.cpp:48] Runtime version 11.10 of cuBLAS incompatible with compiletime version 11.11.
W20230820 16:22:51.365708 798394 cuda_stream.cpp:48] Runtime version 11.10 of cuBLAS incompatible with compiletime version 11.11.
W20230820 16:22:51.380316 798418 cuda_stream.cpp:48] Runtime version 8.4 of cuDNN incompatible with compiletime version 8.6.
W20230820 16:22:51.483361 798396 cuda_stream.cpp:48] Runtime version 8.4 of cuDNN incompatible with compiletime version 8.6.
W20230820 16:22:51.485556 798417 cuda_stream.cpp:48] Runtime version 11.10 of cuBLAS incompatible with compiletime version 11.11.
W20230820 16:22:51.632711 798395 cuda_stream.cpp:48] Runtime version 8.4 of cuDNN incompatible with compiletime version 8.6.
W20230820 16:22:51.669270 798435 cuda_stream.cpp:48] Runtime version 8.4 of cuDNN incompatible with compiletime version 8.6.
W20230820 16:22:51.758263 798393 cuda_stream.cpp:48] Runtime version 8.4 of cuDNN incompatible with compiletime version 8.6.
W20230820 16:22:51.794847 798394 cuda_stream.cpp:48] Runtime version 8.4 of cuDNN incompatible with compiletime version 8.6.
W20230820 16:22:51.936589 798417 cuda_stream.cpp:48] Runtime version 8.4 of cuDNN incompatible with compiletime version 8.6.
Rank[0], Step 1000, Loss nan, Latency 27.498 ms, Throughput 2010891.9, 2023-08-20 16:23:23
Rank[0], Step 2000, Loss nan, Latency 17.224 ms, Throughput 3210378.3, 2023-08-20 16:23:40
Rank[0], Step 3000, Loss nan, Latency 16.778 ms, Throughput 3295727.5, 2023-08-20 16:23:57
Rank[0], Step 4000, Loss nan, Latency 17.068 ms, Throughput 3239748.7, 2023-08-20 16:24:14
Rank[0], Step 5000, Loss nan, Latency 17.225 ms, Throughput 3210256.4, 2023-08-20 16:24:32
Rank[0], Step 6000, Loss nan, Latency 16.831 ms, Throughput 3285441.4, 2023-08-20 16:24:48
Rank[0], Step 7000, Loss nan, Latency 17.101 ms, Throughput 3233504.5, 2023-08-20 16:25:05
Rank[0], Step 8000, Loss nan, Latency 17.438 ms, Throughput 3171003.6, 2023-08-20 16:25:23
Rank[0], Step 9000, Loss nan, Latency 17.039 ms, Throughput 3245191.2, 2023-08-20 16:25:40
Rank[0], Step 10000, Loss nan, Latency 17.339 ms, Throughput 3189073.0, 2023-08-20 16:25:57
Rank[0], Step 10000, AUC -0.01767, Eval_time 14.62 s, AUC_time 1.62 s, Eval_samples 89137152, GPU_Memory 24321 MiB, Host_Memory 23264 MiB, 2023-08-20 16:26:14
Rank[0], Step 11000, Loss nan, Latency 17.114 ms, Throughput 3230980.8, 2023-08-20 16:26:31
Rank[0], Step 12000, Loss nan, Latency 17.141 ms, Throughput 3226008.2, 2023-08-20 16:26:48
Rank[0], Step 13000, Loss nan, Latency 17.024 ms, Throughput 3248050.4, 2023-08-20 16:27:05
Rank[0], Step 14000, Loss nan, Latency 16.490 ms, Throughput 3353375.6, 2023-08-20 16:27:22
Rank[0], Step 15000, Loss nan, Latency 17.255 ms, Throughput 3204596.1, 2023-08-20 16:27:39
Rank[0], Step 16000, Loss nan, Latency 16.219 ms, Throughput 3409284.0, 2023-08-20 16:27:55
Rank[0], Step 17000, Loss nan, Latency 17.324 ms, Throughput 3191856.2, 2023-08-20 16:28:13
Rank[0], Step 18000, Loss nan, Latency 17.235 ms, Throughput 3208351.5, 2023-08-20 16:28:30
Rank[0], Step 19000, Loss nan, Latency 17.412 ms, Throughput 3175821.6, 2023-08-20 16:28:47
Rank[0], Step 20000, Loss nan, Latency 16.913 ms, Throughput 3269357.5, 2023-08-20 16:29:04
Rank[0], Step 20000, AUC -0.01767, Eval_time 11.66 s, AUC_time 1.64 s, Eval_samples 89137152, GPU_Memory 24321 MiB, Host_Memory 23285 MiB, 2023-08-20 16:29:18
Rank[0], Step 21000, Loss nan, Latency 17.504 ms, Throughput 3158977.3, 2023-08-20 16:29:36
Rank[0], Step 22000, Loss nan, Latency 17.316 ms, Throughput 3193431.1, 2023-08-20 16:29:53
Rank[0], Step 23000, Loss nan, Latency 17.295 ms, Throughput 3197182.5, 2023-08-20 16:30:10
Rank[0], Step 24000, Loss nan, Latency 17.587 ms, Throughput 3144163.0, 2023-08-20 16:30:28
Rank[0], Step 25000, Loss nan, Latency 17.439 ms, Throughput 3170906.7, 2023-08-20 16:30:45
Rank[0], Step 26000, Loss nan, Latency 17.193 ms, Throughput 3216169.3, 2023-08-20 16:31:02
Rank[0], Step 27000, Loss nan, Latency 17.348 ms, Throughput 3187422.7, 2023-08-20 16:31:20
Rank[0], Step 28000, Loss nan, Latency 17.380 ms, Throughput 3181678.1, 2023-08-20 16:31:37
Rank[0], Step 29000, Loss nan, Latency 17.424 ms, Throughput 3173590.3, 2023-08-20 16:31:55
Rank[0], Step 30000, Loss nan, Latency 16.547 ms, Throughput 3341777.5, 2023-08-20 16:32:11
Rank[0], Step 30000, AUC -0.01767, Eval_time 11.59 s, AUC_time 1.47 s, Eval_samples 89137152, GPU_Memory 24321 MiB, Host_Memory 23272 MiB, 2023-08-20 16:32:25
Rank[0], Step 31000, Loss nan, Latency 16.170 ms, Throughput 3419664.1, 2023-08-20 16:32:41
Rank[0], Step 32000, Loss nan, Latency 17.151 ms, Throughput 3224127.5, 2023-08-20 16:32:58
Rank[0], Step 33000, Loss nan, Latency 17.123 ms, Throughput 3229308.4, 2023-08-20 16:33:15
Rank[0], Step 34000, Loss nan, Latency 17.260 ms, Throughput 3203673.2, 2023-08-20 16:33:33
Rank[0], Step 35000, Loss nan, Latency 17.266 ms, Throughput 3202530.6, 2023-08-20 16:33:50
Rank[0], Step 36000, Loss nan, Latency 17.425 ms, Throughput 3173403.0, 2023-08-20 16:34:07
Rank[0], Step 37000, Loss nan, Latency 17.289 ms, Throughput 3198409.1, 2023-08-20 16:34:25
Rank[0], Step 38000, Loss nan, Latency 17.175 ms, Throughput 3219515.4, 2023-08-20 16:34:42
Rank[0], Step 39000, Loss nan, Latency 17.423 ms, Throughput 3173692.1, 2023-08-20 16:34:59
Rank[0], Step 40000, Loss nan, Latency 17.296 ms, Throughput 3197119.1, 2023-08-20 16:35:16
Rank[0], Step 40000, AUC -0.01767, Eval_time 11.33 s, AUC_time 1.44 s, Eval_samples 89137152, GPU_Memory 24321 MiB, Host_Memory 23283 MiB, 2023-08-20 16:35:30
Rank[0], Step 41000, Loss nan, Latency 17.413 ms, Throughput 3175479.2, 2023-08-20 16:35:47
Rank[0], Step 42000, Loss nan, Latency 17.311 ms, Throughput 3194303.4, 2023-08-20 16:36:05
Rank[0], Step 43000, Loss nan, Latency 17.286 ms, Throughput 3198979.1, 2023-08-20 16:36:22
Rank[0], Step 44000, Loss nan, Latency 17.271 ms, Throughput 3201625.7, 2023-08-20 16:36:39
Rank[0], Step 45000, Loss nan, Latency 16.605 ms, Throughput 3330098.4, 2023-08-20 16:36:56
Rank[0], Step 46000, Loss nan, Latency 17.056 ms, Throughput 3242020.4, 2023-08-20 16:37:13
Rank[0], Step 47000, Loss nan, Latency 16.129 ms, Throughput 3428323.4, 2023-08-20 16:37:29
Rank[0], Step 48000, Loss nan, Latency 17.466 ms, Throughput 3165871.9, 2023-08-20 16:37:46
Rank[0], Step 49000, Loss nan, Latency 17.148 ms, Throughput 3224542.3, 2023-08-20 16:38:04
Rank[0], Step 50000, Loss nan, Latency 17.539 ms, Throughput 3152775.2, 2023-08-20 16:38:21
Rank[0], Step 50000, AUC -0.01767, Eval_time 11.64 s, AUC_time 1.52 s, Eval_samples 89137152, GPU_Memory 24321 MiB, Host_Memory 23187 MiB, 2023-08-20 16:38:35
Rank[0], Step 51000, Loss nan, Latency 17.578 ms, Throughput 3145669.0, 2023-08-20 16:38:53
Rank[0], Step 52000, Loss nan, Latency 21.118 ms, Throughput 2618430.6, 2023-08-20 16:39:14
Rank[0], Step 53000, Loss nan, Latency 30.792 ms, Throughput 1795775.8, 2023-08-20 16:39:44
Rank[0], Step 54000, Loss nan, Latency 30.774 ms, Throughput 1796840.9, 2023-08-20 16:40:15
Rank[0], Step 55000, Loss nan, Latency 31.071 ms, Throughput 1779677.2, 2023-08-20 16:40:46
Rank[0], Step 56000, Loss nan, Latency 30.764 ms, Throughput 1797409.8, 2023-08-20 16:41:17
Rank[0], Step 57000, Loss nan, Latency 31.024 ms, Throughput 1782356.5, 2023-08-20 16:41:48
Rank[0], Step 58000, Loss nan, Latency 30.567 ms, Throughput 1809023.0, 2023-08-20 16:42:19
Rank[0], Step 59000, Loss nan, Latency 31.180 ms, Throughput 1773439.0, 2023-08-20 16:42:50
Rank[0], Step 60000, Loss nan, Latency 31.467 ms, Throughput 1757270.8, 2023-08-20 16:43:21
Rank[0], Step 60000, AUC -0.01767, Eval_time 18.53 s, AUC_time 2.21 s, Eval_samples 89137152, GPU_Memory 24321 MiB, Host_Memory 23359 MiB, 2023-08-20 16:43:43
Rank[0], Step 61000, Loss nan, Latency 31.226 ms, Throughput 1770810.6, 2023-08-20 16:44:14
Rank[0], Step 62000, Loss nan, Latency 30.569 ms, Throughput 1808909.1, 2023-08-20 16:44:45
Rank[0], Step 63000, Loss nan, Latency 31.168 ms, Throughput 1774113.5, 2023-08-20 16:45:16
Rank[0], Step 64000, Loss nan, Latency 30.363 ms, Throughput 1821153.6, 2023-08-20 16:45:46
Rank[0], Step 65000, Loss nan, Latency 30.924 ms, Throughput 1788105.9, 2023-08-20 16:46:17
Rank[0], Step 66000, Loss nan, Latency 31.286 ms, Throughput 1767439.8, 2023-08-20 16:46:48
Rank[0], Step 67000, Loss nan, Latency 14.880 ms, Throughput 3716249.6, 2023-08-20 16:47:03
Rank[0], Step 68000, Loss nan, Latency 13.291 ms, Throughput 4160461.7, 2023-08-20 16:47:16
Rank[0], Step 69000, Loss nan, Latency 22.644 ms, Throughput 2441941.4, 2023-08-20 16:47:39
Rank[0], Step 70000, Loss nan, Latency 29.506 ms, Throughput 1874073.3, 2023-08-20 16:48:09
Rank[0], Step 70000, AUC -0.01767, Eval_time 16.96 s, AUC_time 1.87 s, Eval_samples 89137152, GPU_Memory 24321 MiB, Host_Memory 23342 MiB, 2023-08-20 16:48:28
Rank[0], Step 71000, Loss nan, Latency 27.504 ms, Throughput 2010439.1, 2023-08-20 16:48:56
Rank[0], Step 72000, Loss nan, Latency 29.516 ms, Throughput 1873399.4, 2023-08-20 16:49:25
Rank[0], Step 73000, Loss nan, Latency 25.467 ms, Throughput 2171275.0, 2023-08-20 16:49:51
Rank[0], Step 74000, Loss nan, Latency 13.374 ms, Throughput 4134636.7, 2023-08-20 16:50:04
Rank[0], Step 75000, Loss nan, Latency 13.285 ms, Throughput 4162170.2, 2023-08-20 16:50:17
Rank[0], Step 75000, AUC -0.01767, Eval_time 9.32 s, AUC_time 1.37 s, Eval_samples 89137152, GPU_Memory 24321 MiB, Host_Memory 23130 MiB, 2023-08-20 16:50:29
*****************************************
Setting OMP_NUM_THREADS environment variable for each process to be 1 in default, to avoid your system being overloaded, please further tune the variable for optimal performance in your application as needed. 
*****************************************

```

## System Information

- What is your OneFlow installation (pip, source, dockerhub):
- OS:
- OneFlow version (run `python3 -m oneflow --doctor`) : 0.9.0+cu118
- Python version : Python 3.9.17
- CUDA driver version: NVIDIA-SMI 450.203.03   Driver Version: 450.203.03   CUDA Version: 11.0 
- GPU models:  v100
- Other info:
